### PR TITLE
Implement more input callbacks

### PR DESCRIPTION
**Pull request by foxnne**
This contains all necessary new input related callbacks/functions to get pixi back working on macos with scrolling, mouse movement, buttons, etc.

- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: MACHAppDelegate.m [hunk 1]
```diff
@@ -39,4 +39,8 @@ - (NSApplicationTerminateReply)applicationShouldTerminate:(NSApplication *)sende
     return NSTerminateCancel;
 }
 
+- (BOOL)applicationShouldTerminateAfterLastWindowClosed:(NSApplication *)sender {
+    return YES;
+}
```
Comment (slimsag) [inline MACHAppDelegate.m:44]: this seems a little sus to me; does it mean that when the user clicks the exit button on the last remaining window, the app won't get a chance to intercept/handle that event to decide if it wants to shutdown?

File: MACHView.m [hunk 2]
```diff
@@ -1,38 +1,214 @@
...
+    __attribute__((objc_direct)) {
+  _flagsChanged_block = flagsChanged_block;
+}
+
+- (void)setBlock_insertText:(void (^)(NSEvent *, uint32_t))insertText_block
+    __attribute__((objc_direct)) {
+  _insertText_block = insertText_block;
+}
+
+- (void)setBlock_magnify:(void (^)(NSEvent *))magnify_block
```
Comment (slimsag) [inline MACHView.m:68]: nice!

File: MACHView.m [hunk 3]
```diff
@@ -1,38 +1,214 @@
...
 
 - (void)keyDown:(NSEvent *)event {
-    if (_keyDown_block) _keyDown_block(event);
+  if (_keyDown_block)
+    _keyDown_block(event);
+
+  [self interpretKeyEvents:@[ event ]];
+}
+
+- (void)insertText:(id)string {
```
Comment (slimsag) [inline MACHView.m:80]: I guess we can implement image copy/paste in pixi now too maybe :)

**Review by slimsag — APPROVED** [review]

File: MACHAppDelegate.m (hunk 1 above)
Comment (rudedogg) [reply MACHAppDelegate.m:44]: - https://developer.apple.com/documentation/appkit/nsapplicationdelegate/applicationshouldterminateafterlastwindowclosed(_:)?language=objc#Discussion

Looks like [`applicationShouldTerminate()`](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/applicationshouldterminate(_:)?language=objc) would still be called. But I'm not sure about returning [`NSTerminateCancel`](https://developer.apple.com/documentation/appkit/nsapplication/terminatereply/terminatecancel?language=objc) there. Makes it sound like the app couldn't close gracefully.

The [`terminate()`](https://developer.apple.com/documentation/appkit/nsapplication/terminate(_:)?language=objc#Discussion) docs look helpful

**Pull request merged**